File: app/src/main/java/com/tangem/di/NavigatorModule.java
package com.tangem.di;

import com.tangem.presentation.dialog.WaitSecurityDelayDialogNew;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
class NavigatorModule {

    @Singleton
    @Provides
    Navigator provideNavigator() {
        return new Navigator();
    }

    @Singleton
    @Provides
    WaitSecurityDelayDialogNew provideWaitSecurityDelayDialogNew() {
        return new WaitSecurityDelayDialogNew();
    }

}

File: app/src/main/java/com/tangem/di/NavigatorModule.kt
package com.tangem.di

import com.tangem.ui.dialog.WaitSecurityDelayDialogNew
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
internal class NavigatorModule {

    @Singleton
    @Provides
    fun provideToastHelper(): ToastHelper {
        return ToastHelper()
    }

    @Singleton
    @Provides
    fun provideWaitSecurityDelayDialogNew(): WaitSecurityDelayDialogNew {
        return WaitSecurityDelayDialogNew()
    }

}